gi: support GVariant boolean -> Ruby

knatsakis opened the pull request.

kou commented:
This pull request includes a merge commit. Please rebase on master and remove it.

kou commented on gobject-introspection/ext/gobject-introspection/rb-gi-argument.c:
```diff
@@ -548,6 +548,10 @@ interface_variant_to_ruby(GVariant *variant)
         rb_value = CSTR2RVAL(value);
+    } else if (g_variant_type_equal(type, G_VARIANT_TYPE_BOOLEAN)) {
+        const gboolean *value;
+	g_variant_get(variant, "b", &value);
```
A tab is used.

knatsakis replied:
Fixed

kou commented on gobject-introspection/ext/gobject-introspection/rb-gi-argument.c:
```diff
@@ -548,6 +548,10 @@ interface_variant_to_ruby(GVariant *variant)
+    } else if (g_variant_type_equal(type, G_VARIANT_TYPE_BOOLEAN)) {
+        const gboolean *value;
+	g_variant_get(variant, "b", &value);
+	rb_value = CBOOL2RVAL(value);
```
`*value`?

knatsakis replied:
Fixed. I didn't do appropriate testing it seems.

kou commented:
Could you provide a test case or description how to test the change by us?

knatsakis commented:
I have removed the merge commit, replaced tabs with spaces for indentation and fixed the wrong use of *value.

I am testing this with Gio::Settings.
You can find a boolean gsettings value with _dconf-editor_ and try to retrieve the value. E.g.:

From a shell:
`gsettings get org.mate.background show-desktop-icons`
`false`
`gsettings get org.mate.background background-fade`
`true`

From ruby:

`[1] pry(main)> require 'gio2'`
`=> true`
`[2] pry(main)> Gio::Settings::new('org.mate.background').get_value('show-desktop-icons')`
`=> false`
`[3] pry(main)> Gio::Settings::new('org.mate.background').get_value('background-fade')`
`=> true`

kou commented:
Thanks.
I've implemented it in different way (we already have implementation!) and added tests for it by `Gio::Settings`.
